aio.api.github/aio/api/github/abstract/api.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any, Type
+
+import gidgethub
+import gidgethub.abc
+import gidgethub.aiohttp
+
+import abstracts
+
+from .commit import AGithubCommit
+from .issues import AGithubIssue, AGithubIssues
+from .iterator import AGithubIterator
+from .label import AGithubLabel
+from .release import AGithubRelease
+from .repo import AGithubRepo
+from .tag import AGithubTag
+
+
+class AGithubAPI(metaclass=abstracts.Abstraction):
+    """Github API wrapper.
+
+    Can be used to access the gidgethub API directly.
+
+    Can also be used to work with repos by calling `self["repo/name"]`.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        self.args = args
+        self.kwargs = kwargs
+
+    def __getitem__(self, k) -> AGithubRepo:
+        """Return a `GithubRepository` for `k`"""
+        # TODO: make this work with user and organization
+        #  and validate `k`
+        return self.repo_class(self, k)
+
+    @cached_property
+    def api(self) -> gidgethub.abc.GitHubAPI:
+        """Gidgethub API."""
+        return self.api_class(*self.args, **self.kwargs)
+
+    @property
+    @abc.abstractmethod
+    def api_class(self) -> Type[gidgethub.abc.GitHubAPI]:
+        """API class."""
+        return gidgethub.aiohttp.GitHubAPI
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def commit_class(self) -> Type[AGithubCommit]:
+        """Github commit class."""
+        raise NotImplementedError
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def label_class(self) -> Type[AGithubLabel]:
+        """Github label class."""
+        raise NotImplementedError
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def issue_class(self) -> Type[AGithubIssue]:
+        """Github issue class."""
+        raise NotImplementedError
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def issues_class(self) -> Type[AGithubIssues]:
+        """Github issues class."""
+        raise NotImplementedError
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def iterator_class(self) -> Type[AGithubIterator]:
+        """Github iterator class.
+
+        Provides both an async iterator and a `total_count` async prop.
+        """
+        raise NotImplementedError
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def release_class(self) -> Type[AGithubRelease]:
+        """Github release class."""
+        raise NotImplementedError
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def repo_class(self) -> Type[AGithubRepo]:
+        """Github repo class."""
+        raise NotImplementedError
+
+    @property  # type:ignore
+    @abstracts.interfacemethod
+    def tag_class(self) -> Type[AGithubTag]:
+        """Github tag class."""
+        raise NotImplementedError
+
+    async def getitem(self, *args, **kwargs) -> Any:
+        """Call the `gidgethub.getitem` api."""
+        # print("GETITEM", args, kwargs)
```

Comment:
Do we need these commented lines?

Reply:
hmm, in these cases i minded to leave them 

it would probs be better to have debug logging here, but because of the way the checker works that is less straightforward, so ive just got some commented out prints for now